### PR TITLE
fix: rakuten amount value rounded up to nearest integer

### PR DESCRIPTION
**Pull request by shrouti1507**
## What are the changes introduced in this PR?

Write a brief explainer on your code changes.

## What is the related Linear task?

Resolves INT-2737

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

@coderabbitai review

<hr>

### Developer checklist

- [ ] My code follows the style guidelines of this project

- [ ] **No breaking changes are being introduced.**

- [ ] All related docs linked with the PR?

- [ ] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [ ] Is the PR limited to 10 file changes?

- [ ] Is the PR limited to one linear task?

- [ ] Are relevant unit and component test-cases added in **new readability format**?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3784/test-report.html

File: src/cdk/v2/destinations/rakuten/utils.js [hunk 1]
```diff
@@ -17,6 +17,18 @@ const constructProperties = (message) => {
...
+/**
+ * Calculates the amount for a single product
+ * @param {Object} product
+ * @returns {number}
+ */
+const calculateProductAmount = (product) => {
+  if (!product?.amount && !product?.price) {
+    throw new InstrumentationError('Either amount or price is required for every product');
+  }
+  return Math.round(product.amount) * 100 || (product.quantity || 1) * 100 * product.price;
```
Comment (sanpj2292) [inline src/cdk/v2/destinations/rakuten/utils.js:29]:
```suggestion
  return Math.round(product.amount) * 100 || (product.quantity || 1) * Math.round(100 * product.price);
```
Should we also do ?

Comment (koladilip) [reply src/cdk/v2/destinations/rakuten/utils.js:29]: return Math.round(product.amount * 100 || (product.quantity || 1) * 100 * product.price);

Comment (koladilip) [reply src/cdk/v2/destinations/rakuten/utils.js:29]: we should do Math.round after multiplying with 100 right?

Comment (shrouti1507) [reply src/cdk/v2/destinations/rakuten/utils.js:29]: yes

Comment (shrouti1507) [reply src/cdk/v2/destinations/rakuten/utils.js]: yes yes, changed

File: src/cdk/v2/destinations/rakuten/utils.js [hunk 2]
```diff
@@ -52,14 +64,8 @@ const constructLineItems = (properties) => {
   });
 
   // Map 'amountList' by evaluating 'amount' or deriving it from 'price' and 'quantity'
-  const amountList = products.map((product) => {
-    if (!product?.amount && !product?.price) {
-      throw new InstrumentationError('Either amount or price is required for every product');
-    }
-    return product.amount * 100 || (product.quantity || 1) * 100 * product.price;
```
Comment (krishna2020) [inline src/cdk/v2/destinations/rakuten/utils.js:59]: If both amount & price are in the payload, Is this calculation considering both amount & price is valid ? cc: @anantjain45823

Comment (anantjain45823) [reply src/cdk/v2/destinations/rakuten/utils.js:59]: Yes we are giving priority to amount over qty * price

Comment (krishna2020) [reply src/cdk/v2/destinations/rakuten/utils.js:59]: sorry, I've overseen the or condition

**Review by krishna2020 — APPROVED** [review]

**Pull request merged**